src/main/java/core/basesyntax/service/strategy/strategyimpl/ReturnOperation.java

```diff
@@ -0,0 +1,17 @@
+package core.basesyntax.service.strategy.strategyimpl;
+
+import core.basesyntax.dao.Storage;
+import core.basesyntax.model.FruitRecord;
+
+public class ReturnOperation implements OperationHandler {
+    @Override
+    public void apply(FruitRecord transaction) {
+        int returnFruits = transaction.getQuantity();
+        Storage.storage.merge(transaction.getFruit(),returnFruits, Integer::sum);
+
+        if (returnFruits < 0) {
+            throw new RuntimeException("Balance cannot be negative for fruit: "
+                    + transaction.getFruit());
+        }
+    }
```

Comment:
The class `ReturnOperation` should be placed in the `strategy` package, not in `strategy.strategyimpl`. According to the checklist, the project structure should consist of a `strategy` package for holding handlers for different operations.